### ISSUE TITLE
GODS 0.9B doesn't launch

### ISSUE THREAD

**Issue by nabla666** [jotd666/gods-deluxe#1]
Gods 0.9B doesn't launch whereas Gods 0.9 is working fine.
I have java 8 v271 x64 (build 1.8.0_271-b09) under W10 x64 20H2.
![JNI Error](https://user-images.githubusercontent.com/18247053/98445130-4d9c0600-2116-11eb-8fb6-637535faac60.png)

**Comment by jotd666** [comment]
hi,

probably a joystick.dll issue. Can you delete this dll and retry? joystick
won't work but at least it will launch

**Comment by nabla666** [comment]
**I deleted the joystick.dll but the problem persists.**
( The good new is that this dll make the 0.9 version work with the joystick ;)

**Comment by jotd666** [comment]
ah that's more logical, as I recently rebuilt the DLL for 64 bit
architecture, which is now the standard. Thanks for this report, I will
investigate

**Comment by CGJennings** [comment]
I just downloaded the prebuilt version from your Web site. The class file version is 57, meaning it was compiled for Java 13 or newer. It won't run under Java 8, which can use up to version 52. I should run under Java 13+. I am running it with Java 16 and it seems to be working fine.

**Comment by nabla666** [comment]
I have just tried to install JDK 17 and it seems to work !

**Comment by CGJennings** [comment]
Glad it worked for you! Seems like an easy fix then, either recompile for Java 8 or note the minimum version requirement.

**Comment by jotd666** [comment]
seriously those java versions are driving me insane. I more and more regret not having coded this game in C++... thanks for helping each other on my software :)

**Comment by ingomohr** [comment]
Java 8 is from 2014. Current LTS is Java 17. LTS before that was Java 11.
I would think twice before I would take any action to support sich an outdated version as Java 8 is.

**Comment by jotd666** [comment]
Yes, I don't care much for old java versions. Any one of you wants to create a pull request on anything including the readme to explain that? or the issue is that the binary package of my webpage only runs with old java 8 version?

**Comment by CGJennings** [comment]
The issue is that the binary package on your page requires Java 13+, because that is what was used to compile it. As far as I know, the code in the repo still compiles fine for Java 8, though I haven't checked. Unless you change the source code to actually use Java 9+ features, there is no real reason to exclude Java 8. Java 8 is still actively updated and supported (for free) by a number of corporations and open source groups. (It is widely used on servers and in enterprise.) Given that it gets fixes without getting new features, it is arguably *more* stable than any version that has come after.

The easiest fix is to leave this repo unchanged and just add a note to your download page stating that it requires the latest version of Java (or 13+ if you want to be precise).

**Comment by mbugni** [comment]
Please give a try to [Release 0.9.0](https://github.com/jotd666/gods-deluxe/releases/tag/0.9.0). It requires Java 8 as minimum.

**Comment by mbugni** [comment]
Version [0.10.0 is available](https://github.com/jotd666/gods-deluxe/releases/tag/0.10.0). It requires Java 11 or above. Please let me know if this issue is still present.
